src/services/conversionRates/getConversionRatesService.js

```diff
@@ -50,45 +50,18 @@ const _getRatesCoinGecko = async (
   const promises = ratesToGet.map(async r => {
     const rateSymbolInner = stableCoins.includes(r) ? 'USD' : r;
     if (rateSymbolInner !== rateSymbol) {
-      const today = new Date(new Date());
-      const todayUTC = today.setUTCHours(0, 0, 0, 0);
-      const todayRounded = Math.round(todayUTC / 1000);
-      const timestampRounded = Math.round(timestampMS / 1000);
-      const differenceMillis = todayRounded - timestampRounded;
-      let differenceDays = 0;
+      const timestampTo = Math.round(timestampMS / 1000);
+      const timestampFrom = timestampTo - 3600 * 12;
       let bestPrice = 1;
-      if (differenceMillis <= 0) {
-        differenceDays = 0;
-      } else {
-        differenceDays = differenceMillis / 86400;
-      }
-
-      const daysMax = Math.round(differenceDays);
       const testRep = JSON.parse(
         await rp(
-          `https://api.coingecko.com/api/v3/coins/${coingeckoId}/market_chart?vs_currency=${rateSymbolInner}&days=${daysMax}`,
+          `https://api.coingecko.com/api/v3/coins/${coingeckoId}/market_chart/range?vs_currency=${rateSymbolInner}&from=${timestampFrom}&to=${timestampTo}`,
         ),
       );
 
       if (testRep) {
-        let difference = 0;
-        let bestIndex = 0;
-        let bestDifference = Infinity;
-        let i;
-        let cur;
-        let priceTime;
-
         const { prices } = testRep;
-        for (i = 0; i < prices.length; i += 1) {
-          cur = prices[i];
-          priceTime = Math.round(cur[0] / 1000);
-          difference = Math.abs(timestampRounded - priceTime);
-          if (difference < bestDifference) {
-            bestDifference = difference;
-            bestIndex = i;
-          }
-        }
-        const bestPrices = prices[bestIndex];
+        const bestPrices = prices[0];
```

Comment:
The first price returned is not the best (closest one). Iterate over them to find the best one.

Reply:
Re implemented timeframe refination